File: 401-challenges/algorithms/arrayShift/array_shift.test.js
'use strict';
const insertShiftArray = require('./array_shift.js');

describe('Testing [2,4,6,8] with 5', () => {
  test('It should return [2,4,5,6,8]', () => {
    expect(insertShiftArray([2,4,6,8], 5)).toStrictEqual([2,4,5,6,8]);
  });
});

describe('Testing [4,8,15,23,42] with 16', () => {
  test('It should return [4,8,15,16,23,42]', () => {
    expect(insertShiftArray([4,8,15,23,42], 16)).toStrictEqual([4,8,15,16,23,42]);
  });
});

describe('Testing [1,2,3,5,6,7] with 4)', () => {
  test('It should return [1,2,3,4,5,6,7]', () => {
    expect(insertShiftArray([1,2,3,5,6,7], 4)).toStrictEqual([1,2,3,4,5,6,7]);
  });
});


describe('Testing [1,2,4] with 3)', () => {
  test('It should return [1,2,3,4]', () => {
    expect(insertShiftArray([1,2,4], 3)).toStrictEqual([1,2,3,4]);
  });
});

File: 401-java-challenges/lib/src/test/java/challenges/ArrayShiftTest.java
package challenges;
import org.junit.Test;
import java.lang.reflect.Array;
import java.util.ArrayList;
import static org.junit.Assert.*;

public class ArrayShiftTest {
    @Test public void testArrayShift1() {
        int[] testArr = {2,4,6,8};
        int[] expectedArr = {2,4,5,6,8};
        ArrayList expected = listGenerator(expectedArr);
        ArrayList actual = ArrayShift.insertShiftArray(listGenerator(testArr), 5);
        assertEquals("two arrives should be the same", expected, actual);
    }

    @Test public void testArrayShift2() {
        int[] testArr = {4,8,15,23,42};
        int[] expectedArr = {4,8,15,16,23,42};
        ArrayList expected = listGenerator(expectedArr);
        ArrayList actual = ArrayShift.insertShiftArray(listGenerator(testArr), 16);
        assertEquals("two arrives should be the same", expected, actual);
    }

    private ArrayList listGenerator (int[] arr){
        ArrayList<Integer> result = new ArrayList<>();
        for (int val : arr){
            result.add(val);
        }
        return result;
    }
}
